src/components/editor-page/hooks/useUpdateHistoryEntry.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import equal from 'fast-deep-equal'
+import { useCallback, useEffect, useRef } from 'react'
+import { useSelector } from 'react-redux'
+import { ApplicationState, store } from '../../../redux'
+import { useParams } from 'react-router-dom'
+import { EditorPagePathParams } from '../editor-page'
+import { HistoryEntry, HistoryEntryOrigin } from '../../../redux/history/types'
+import { updateLocalHistoryEntry } from '../../../redux/history/methods'
+
+export const useUpdateHistoryEntry = (loading: boolean, error: boolean): void => {
+  const { id } = useParams<EditorPagePathParams>()
+  const userExists = useSelector((state: ApplicationState) => !!state.user)
+  const noteTitle = useSelector((state: ApplicationState) => state.noteDetails.noteTitle)
+  const noteTags = useSelector((state: ApplicationState) => state.noteDetails.frontmatter.tags)
+
+  const lastTitle = useRef('')
+  const lastTags = useRef<string[]>([])
+
+
+  const updateHistory = useCallback(() => {
+    if (loading || error || userExists) {
+      return
+    }
+    // This is needed to not update the history entry on each scroll without changes
+    // just because title or tags redux state were touched.
```

Comment:
You don't need this comment if you name your variables correctly. If I see a comparison of `lastNoteTitle` with e.g. `currentNoteTitle` then it's clear enough.

Reply:
Changed.